pkg/util/containers/metrics/system/collector_linux.go

```diff
@@ -202,6 +205,23 @@ func (c *systemCollector) GetContainerIDForPID(pid int, cacheValidity time.Durat
 	return containerID, err
 }
 
+func (c *systemCollector) GetContainerIDForInode(inode uint64, cacheValidity time.Duration) (string, error) {
+	cg := c.reader.GetCgroupByInode(inode)
+	if cg == nil {
+		err := c.reader.RefreshCgroups(cacheValidity)
```

Comment:
Is `RefreshCgroups` or `GetCgroupByInode` call possibly slow? Linked to the DogStatsD parsing code, `GetContainerIDForInode ` is part of a hotpath.

Reply:
I guess it depends on what you call slow. It needs to perform IO. IIRC it's reading every `/proc/<pid>/cgroup` file

Reply:
Sounds good! Thanks